data/phones/arm_e_phonetic.phones

```diff
@@ -1,66 +1,94 @@
 # Based on https://en.wikipedia.org/wiki/Armenian_language#Phonology
+# And based on the pronunciation script from Wiktionary: https://en.wiktionary.org/wiki/Module:hy-pronunciation
+#
 # Vowels.
 ɑ
 ɛ
 ə
 i
 ɔ
-o  # <ո> is really [ɔ].
-a  # <ա> is really [ɑ].
 u
+#
+# Older entries might contain [o] or [e]. These have been fixed. You shouldn't find these
+# phones anymore.
+#
+# Long vowels: A sequence of identical heterosyllabic vowels is phonetically pronounced as 
+# one long vowel. Armenian does not have phonemic vowel length. But these are very rare. 
+# The only attested case on Wiktionary is the following.
+ɛː
+# The other types of long vowels are not found on Wiktionary as of Jan 2021. This is 
+# largely because long vowels are mostly in loanwords and borrowings, which Wiktionary 
+# underreports. At some point in the future, the Armenian users might add some entries 
+# which contain these sequences.
+# ɑː əː iː ɔː uː
+#
+#
 # Consonants.
 m
-mː
 n
-nː
 ŋ
 pʰ
-pʰː
 p
-pː
 b
-bː
 tʰ
-tʰː
 t
-tː
 d
-dː
 kʰ
-kʰː
 k
-kː
 ɡ
-ɡː
 t͡sʰ
-t͡sʰː
-t͡s  # No corresponding geminate?
-d͡z  # No corresponding geminate?
+t͡s
+d͡z
 t͡ʃʰ
 t͡ʃ
-d͡ʒ  # No corresponding geminate?
+d͡ʒ
 f
-fː
 v
-vː
 s
-sː
 z
-zː
 ʃ
-ʃː
-ʒ  # No corresponding geminate?
-χ  # Sometimes [x]?
-χː
+ʒ
+χ  
 ʁ
-ʁː
-h  # No corresponding geminate?
+h
 l
-lː
-j  # No corresponding geminate?
+j
 r
 ɾ
+#
+# Past errors: some instances of [χ] were incorrectly transcribed as [x]. 
+# Past errors: some affricates were missing a tie-bar. 
+#  * <ց> [tʃʰ] 
+#  * <չ> [tsʰ]
+# These were fixed.
+# 
+# Long consonants: A sequence of identical consonants are phonetically pronounced as 
+# long consonant. Armenian does not have phonemic consonant length or geminates.
+mː
+nː
+pʰː
+pː
+bː
+tʰː
+tː
+dː
+kʰː
+kː
+ɡː
+t͡sʰː
+fː
+vː
+sː
+zː
+ʃː
+χː
+ʁː
+lː
 ɾː
-# TODO: for future cleanup, missing tie-bars on segments like:
-# * <ց> [tʃʰ]
-# * <չ> [tsʰ]
+#
+# As an accidental gap,  as of Jan 2021, there are no Wiktionary entries that have the
+# following long consonants:
+# t͡sː d͡zː d͡ʒː ʒː hː jː
```

Comment:
In other languages on Wiktionary, I have seen geminate affricates transcribed as /tts/ or /t:s/ instead of /t͡sː/ (to take one example). Also, if an affricate is transcribed without a tie bar, then our script treats it as a sequence of multiple segments, rather as one sound. For example, if /ts/ is transcribed as such, without a tie bar, then it will be segmented as the sequence /t/+/s/, rather than as a single affricate. Sometimes, this can make it look like affricates are missing from the data when they're actually present.

So my comment here is, have you checked that the missing long affricates aren't being transcribed in a "non-standard" way?

Reply:
Yup. Many of the early entries were manually typed without using the tie-bar diagraph. I converted all those into entries that use the hy-pron script which automatically converts the affricate letters into the right affricate IPA segments with tie-bars.
For geminate affricates, they all look like [t͡sʰː]